cookbooks/swift/recipes/configs.rb

```diff
@@ -265,52 +263,128 @@
     group node["username"]
     action :create
   end
-  template "/#{service_dir}/default.conf-template" do
-    source "#{service_dir}/default.conf-template.erb"
+  template "/#{service_dir}/server.conf-template" do
+    source "#{service_dir}/server.conf-template.erb"
+    owner node["username"]
+    group node["username"]
+    variables({
+      :zipkin => if node["zipkin"] then "zipkin" else "" end,
+      :servers_per_port => node['servers_per_port'],
+      :replication_server => if node["replication_servers"] then "False" else "True" end,
+    })
+  end
+  if node["replication_servers"] then
+    template "/#{service_dir}/replication-server.conf-template" do
+      source "#{service_dir}/server.conf-template.erb"
+      owner node["username"]
+      group node["username"]
+      variables({
+        :zipkin => if node["zipkin"] then "zipkin" else "" end,
+        :servers_per_port => node['servers_per_port'],
+        :replication_server => "True",
+      })
+    end
+  else
+    file "/#{service_dir}/replication-server.conf-template" do
+      action :delete
+    end
+  end
+  template "/#{service_dir}/replication-daemons.conf-template" do
+    source "#{service_dir}/replication.conf-template.erb"
     owner node["username"]
     group node["username"]
-    variables(service_vars[service])
+    variables({
+      :auto_shard => node['container_auto_shard'],
+      :sync_method => node['object_sync_method'],
+    })
   end
   (1..node['nodes']).each do |i|
-    bind_ip = "127.0.0.1"
     bind_port = "60#{i}#{p}"
+    replication_bind_port = "60#{i + 4}#{p}"
     if service == :object && node['servers_per_port'] > 0 then
-      # Only use unique IPs if servers_per_port is enabled.  This lets this
-      # newer vagrant-swift-all-in-one work with older swift that doesn't have
-      # the required whataremyips() plumbing to make unique IPs work.
-      bind_ip = "127.0.0.#{i}"
-
-      # This config setting shouldn't really matter in the server-per-port
-      # scenario, but it should probably at least be equal to one of the actual
+      # These config settings shouldn't really matter in the server-per-port
+      # scenario, but they should probably at least be equal to one of the actual
       # ports in the ring.
       bind_port = "60#{i}6"
+      replication_bind_port = "60#{i + 4}6"
     end
-    conf_dir = "#{service_dir}/#{i}.conf.d"
-    directory "/#{conf_dir}" do
+    server_conf_dir = "#{service_dir}/#{i}.conf.d"
+    replication_conf_dir = "#{service_dir}/#{i+4}.conf.d"
```

Comment:
I'd actually started out with that, but ran into trouble with probe tests and the numbered configs going

1. data
2. replication
3. data
4. replication
 ...

instead of

1. data
2. data
...
5. replication
6. replication
...

I think we'll still need the offset, but add `-replication` seems like a good call.

Reply:
I strongly prefer the interleaved setup - it feels more like simulating 4 nodes with 2 configs each - instead of 8 servers.

... but I haven't tried to make probetests pass with such a config - I supposed if I did, I might come around to "living with it" in the interest of expediency.   I know it seems minor but the ongoing cognitive load of having to add 3+4=7 every-time I want to stop the "the replication server for /srv/node3" seems significant compared to the one time cost of making probetests more expressive/configurable.  :man_shrugging:

Reply:
Well, good news is `swift-init`'s magic helps a lot:
```
vagrant@saio:~/swift$ swift-init status container -c 3
container-server running (9465 - /etc/swift/container-server/3.conf.d)
vagrant@saio:~/swift$ swift-init status container.replication -c 3
container-server running (9469 - /etc/swift/container-server/7-replication.conf.d)
```